pkg/disk/filesystem.go

```diff
@@ -9,47 +9,106 @@ import (
 	"github.com/google/uuid"
 )
 
-type MkfsOption int
+type MkfsOption interface {
```

Comment:
A maybe silly question - if we would just do:
```go
type MkfsOptions struct {
	Verity bool `json:"verity,omitempty" yaml:"verity,omitempty"`

	Heads           int `json:"heads,omitempty" yaml:"heads,omitempty"`
	SectorsPerTrack int `json:"sectors-per-track,omitempty" yaml:"sectors-per-track,omitempty"`
}
```
here instead of going the interface approach, what would be loose? It would be simpler to read and more inline with the rest of the disk code which mostly uses similar (simple) structs. But maybe I'm missing something?

(full diff with minor test failure here: 
[pr-1825-struct-only-approach.diff.txt](https://github.com/user-attachments/files/22119987/pr-1825-struct-only-approach.diff.txt)

Reply:
I'd still like a Geometry struct to show that these are related and you must either specify none or both. But for sure this is nicer. It will change how the json is serialized, but i don't think anything is tied to the existing format, right?

Reply:
done